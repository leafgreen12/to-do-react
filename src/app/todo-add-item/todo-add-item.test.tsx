import React, {useContext} from 'react';
import {act, render, screen} from '@testing-library/react';
import {unmountComponentAtNode} from 'react-dom';
import Enzyme, {shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';
import TodoAddItem from "./todo-add-item";

// @ts-ignore
(Enzyme as any).configure({adapter: new Adapter()})
describe('TodoAddItem', () => {
    let container = null as any;
    let todoAddItem = null as any;
    beforeEach(() => {
        // cài đặt một DOM element như là target cho render
        container = document.createElement("div");
        document.body.appendChild(container);
        todoAddItem = shallow(<TodoAddItem dispatch={(action: any) => null} onSubmit={(action: any) => null}/>)
    });

    afterEach(() => {
        // dọn dẹp lúc thoát
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    it('should render todoAddItem correctly', () => expect(todoAddItem).toMatchSnapshot());
});
