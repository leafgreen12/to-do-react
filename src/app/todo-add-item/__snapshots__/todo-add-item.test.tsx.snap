// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoAddItem should render todoAddItem correctly 1`] = `ShallowWrapper {}`;
